### DISCUSSION TITLE
Is it possible to make directus infer the fields and relationships without having to explicitly add such data?

### DISCUSSION THREAD

**Discussion by wasims1** [directus/directus#17705] [Feature Requests]
### Summary

Make Directus automatically infer the fields, relationships and manage them

### Motivation

Ease of implementing Directus in already established projects

### Detailed Design

For projects that already have an established database and would like to utilize Directus. Is it possible to have Directus read all the fields and the relationships automatically (i believe relationships can be inferred using foreign keys...). This would have several advantages: 
- Increase trust and utility of Directus as it would minimize the time/effort investment in utilizing it
- Makes migration tracking much easier as you can do migrations using Knex for example, and Directus would automatically see the new updates
- Unify field and data management. Currently, first Directus has to manage the field. Then, you have to set who is authorized to access/edit/delete... However if Directus is automatically managing all, you can control all that using one interface which is the authorization interface...

Or maybe there's a simple way to utilize Directus in a large database without having to add all these fields and relationships manually that I don't know about?

### Must Have:
- A Read fields (already done)
- B Infer relationships
- C Assume management of both fields and relationships

### Drawbacks

- It would just change how directus start, from having to tell directus what fields and relationships to manage to having it manage everything....

### Alternatives

- either write the whole migrations using directus (new projects), or have to add fields/relationships to tell directus what to manage (costly)

### Adoption Strategy

I would think it should be easily adopted

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
